File: factory/typescript/concreteSensors/temperature.ts
/** @format */

import { SensorOperations } from "../gateway";

export class Temperature implements SensorOperations {
  readSensorValue(): number {
    return Math.random() * 10;
  }
  receiveCommand(command: string): void {
    console.log(`implement this command ${command}`);
  }
}

File: factory/golang/domain/concreteSensors/temperature.go
package concretesensors

import (
	"factory/domain"
	"fmt"
)

type TemperatureSensor struct {
	Name string
}

func ConcreteTemperatureSensor() domain.SensorOperations {
	return &TemperatureSensor{
		Name: "DHT22",
	}
}

func (s *TemperatureSensor) ReadSensorValue() float32 {
	return 30.0
}

func (s *TemperatureSensor) ReceiveCommand(command string) {
	fmt.Println(fmt.Sprintf("temperature %s", command))
}
